x-pack/plugins/apm/server/routes/source_maps/route.ts

```diff
@@ -91,7 +114,7 @@ const uploadSourceMapRoute = createApmServerRoute({
   endpoint: 'POST /api/apm/sourcemaps 2023-10-31',
   options: {
     tags: ['access:apm', 'access:apm_write'],
-    body: { accepts: ['multipart/form-data'] },
+    body: { accepts: ['multipart/form-data'], maxBytes: 30 * 1024 * 1024 },
```

Comment:
This increases the default limit from 1mb to 30mb. What if the user wants to increase it even further to say 60mb?

Reply:
It might be okay to just have a hardcoded limit for now. Pretty sure someone will eventually complain that their source maps are larger than our limit and then we won't have any workarounds for them. 

[I've asked core](https://elastic.slack.com/archives/C5TQ33ND8/p1688638215610649?thread_ts=1688420450.099789&cid=C5TQ33ND8) so if we can create an escape hatch for users that would be great. If not let's just leave it at this

Reply:
Got it, it was my understanding that they were able to configure it if they wanted to, however, if that's not the case then I think we should definitely set a bigger limit. 30mb was a value that I found to be a bit above the average for Android maps, although that doesn't mean they can't be bigger. So maybe to be sure we don't cause issues for the people using this feature, we could set some limit that seems very unlikely to get reached, such as 100mb for example.

Reply:
> we could set some limit that seems very unlikely to get reached, such as 100mb for example.

Sounds good to me. The route has the access tags making it less likely that someone can abuse it. 

@elastic/kibana-core Any objections to this?

Reply:
As long as the route requires authentication, I don't have any.

Reply:
Thanks @pgayvallet - It does require authentication, so I've just applied the changes. In terms of the server-side memory needed, what would be your recommendation on how to calculate that value? I'm currently not aware of the server details on that matter.

Reply:
I think @pgayvallet mostly mention it to make sure we keep it in mind. At least it made me reconsider some things (transformations) we are doing.

Reply:
Yeah, that's what I meant!

Reply:
Cool! Thanks 👍